tcf_website/views/browse.py

```diff
@@ -148,14 +148,70 @@ def course_instructor(request, course_id, instructor_id):
 def instructor_view(request, instructor_id):
     """View for instructor page, showing all their courses taught."""
     instructor = Instructor.objects.get(pk=instructor_id)
-    avg_rating = instructor.average_rating()
-    if avg_rating is not None:
-        avg_rating = round(avg_rating, 2)
-    avg_difficulty = instructor.average_difficulty()
-    if avg_difficulty is not None:
-        avg_difficulty = round(avg_difficulty, 2)
+
+    avg_rating = safe_round(instructor.average_rating())
+    avg_difficulty = safe_round(instructor.average_difficulty())
+
+    courses = group_by_dept(instructor, instructor.get_courses())
     return render(
         request, 'instructor/instructor.html', {
             'instructor': instructor,
             'avg_rating': avg_rating,
-            'avg_difficulty': avg_difficulty})
+            'avg_difficulty': avg_difficulty,
+            'courses': courses
+        })
+
+
+def group_by_dept(instructor, courses):
+    """ Group instructor's courses by subdepartment.
+
+        Returns a dictionary mapping subdepartment names to ids and
+        lists of Course data.
+    """
+    subdept_ids = list(
+        set(courses.values_list('subdepartment', flat=True)))
+
+    # Contains IDs and names of all subdepartments for Instructor's Courses
+    subdepts = sorted(
+        list(
+            map(
+                lambda subdept_id: [
+                    subdept_id,
+                    Subdepartment.objects.get(
```

Comment:
I think since each professor is likely to only teach in a few subdepartments, this'll only run at most 3-4 times, so it's not urgent to fix, but I'll throw it on the backlog in case we ever wanna take a crack at it.

Reply:
Issue #210